wp-content/plugins/wporg-learn/inc/blocks.php

```diff
@@ -24,10 +24,63 @@
  * @return void
  */
 function register_types() {
+	register_lesson_plan_details();
 	register_workshop_details();
 	register_workshop_application_form();
 }
 
+/**
+ * Register Lesson Plan Details block type and related assets.
+ *
+ * @throws Error If the build files are not found.
+ */
+function register_lesson_plan_details() {
+	$script_asset_path = get_build_path() . 'lesson-plan-details.asset.php';
+	if ( ! is_readable( $script_asset_path ) ) {
+		throw new Error(
+			'You need to run `npm start` or `npm run build` for the "wporg-learn/lesson-plan-details" block first.'
+		);
+	}
+
+	$script_asset = require $script_asset_path;
+	wp_register_script(
+		'lesson-plan-details-editor-script',
+		get_build_url() . 'lesson-plan-details.js',
+		$script_asset['dependencies'],
+		$script_asset['version']
+	);
+
+	wp_register_style(
+		'lesson-plan-details-style',
+		get_build_url() . 'style-lesson-plan-details.css',
+		array(),
+		filemtime( get_build_path() . 'style-lesson-plan-details.css' )
+	);
+
+	register_block_type( 'wporg-learn/lesson-plan-details', array(
+		'editor_script'   => 'lesson-plan-details-editor-script',
+		'style'           => 'lesson-plan-details-style',
+		'render_callback' => __NAMESPACE__ . '\lesson_plan_details_render_callback',
+	) );
+}
+
+/**
+ * Render the block content (html) on the frontend of the site.
+ *
+ * @param array  $attributes
+ * @param string $content
+ * @return string HTML output used by the block
+ */
+function lesson_plan_details_render_callback( $attributes, $content ) {
+	$post    = get_post();
+	$details = wporg_learn_get_lesson_plan_taxonomy_data( get_the_ID(), 'single' );
```

Comment:
This function exists in [functions.php](https://github.com/WordPress/Learn/blob/trunk/wp-content/themes/pub/wporg-learn-2020/functions.php#L230) and it doesn't feel quite right to me for the plugin to depend on it. The function is also called by [the code which renders the details in Lesson Plan archive cards](https://github.com/WordPress/Learn/blob/trunk/wp-content/themes/pub/wporg-learn-2020/archive-lesson-plan.php#L34) so it can't be moved. Should it duplicated in the plugin, or is there prior art for explicit code sharing between the theme and a plugin?

Reply:
@tellyworth @ryelle I'd like your input on this please 🙏

Reply:
I think we tend to go the other way — functions defined in plugins can be used in themes. That feels more natural to me, since a theme is more likely to be changed (in a redesign), but the data structure would stay the same.

I wouldn't consider this a blocker right now, though, since you know the theme shouldn't change.